### ISSUE TITLE
Unable to Modify RLC with Validating Webhook Enabled

### ISSUE THREAD

**Issue by kevin-shelaga** [solo-io/gloo#8042]
### Gloo Edge Version

1.13.x (latest stable)

### Kubernetes Version

None

### Describe the bug

With the validating webhook enabled, changing the `requestsPerUnit` on a RLC and applying the change causes the config to be rejected with errors.

```sh
failed to process RateLimitConfig resource with name [global-limit] in namespace [production]: unknown rate limit config type <nil> 
```

### Steps to reproduce the bug

1. enabled validating webhook
2. apply the following config and reference it to a VS
```
apiVersion: ratelimit.solo.io/v1alpha1
kind: RateLimitConfig
metadata:
  name: global-limit
  namespace: staging
spec:
  raw:
    descriptors:
    - key: generic_key
      value: counter
      rateLimit:
        requestsPerUnit: 450
        unit: MINUTE
    rateLimits:
    - actions:
      - genericKey:
          descriptorValue: counter
```
3. Update the `requestsPerUnit`
```
apiVersion: ratelimit.solo.io/v1alpha1
kind: RateLimitConfig
metadata:
  name: global-limit
  namespace: staging
spec:
  raw:
    descriptors:
    - key: generic_key
      value: counter
      rateLimit:
        requestsPerUnit: 15000
        unit: MINUTE
    rateLimits:
    - actions:
      - genericKey:
          descriptorValue: counter
```

### Expected Behavior

- [ ] No errors
- [ ] All related issues (see below) are confirmed to be resolved as well

### Additional Context

Workaround: Remove the RLC from being referenced by the VS, make the change and apply, then reference again. Or simply disable the validating webhook and make your change, then enable again.

### Related Issues
There are a few issues which are either duplicates of this, or very similar, so we have closed them, and expect them to be solved by this work:
- https://github.com/solo-io/gloo/issues/8238
- https://github.com/solo-io/gloo/issues/8239

**Comment by nfuden** [comment]
reopening for backports

**Comment by sam-heilbron** [comment]
https://github.com/solo-io/gloo/issues/8239 is identical I believe. We should be able to close once this is resolved, but we should double check

**Comment by sam-heilbron** [comment]
I think that the following issues are all identical:
 - [Unable to change requestsPerUnit on RLC with validating webhook enabled](https://github.com/solo-io/gloo/issues/8042) (this one)
 - [rateLimitConfig with Nested Descriptors fail](https://github.com/solo-io/gloo/issues/8238#top)
 - [rateLimitConfig doesn't allow quotes in the value for key: type #8239](https://github.com/solo-io/gloo/issues/8239)
 
 
 They each are based around issues with validating RateLimitConfigurations, all due (I believe) to the same underlying issue. I think when we solve 1, it will solve the rest, but we should also double check. I would be in favor of just collapsing the 3 issues into 1 (closing the others as duplicates) so it's easier to track

**Comment by elcasteel** [comment]
In the course of investigating this, I also discovered that Gloo will accept ratelimit configurations that are later rejected by Envoy - specifically Envoy expects descriptors to have values but gloo doesn't require it.

**Comment by elcasteel** [comment]
This was already fixed in 1.15.
The backports are in OSS and will be in gloo-ee next time we bump dependencies (presumably 1.14.9 and 1.13.24)
